src/nccl_ofi_net.c

```diff
@@ -1050,49 +1064,130 @@ static inline ncclResult_t nccl_ofi_progress(nccl_ofi_t *nccl_ofi_comp)
 }
 
 /*
- * @brief	Checks if the underlying hardware type matches the input platform type.
+ * @brief	Provides EC2 platform type as reported by the
+ * 		first line of
+ *		/sys/devices/virtual/dmi/id/product_name.
+ *		Users of this API *should* free the buffer when a
+ *		Non-NULL string is returned.
  *
- * @return	-1, on error
- *		1, if it matches
- *		0, if it does not match
+ * @return	NULL, on allocation and file system error
+ * 		EC2 platform type, on success
  */
-static int platform_type_matches(char *platform_type)
+static char* get_platform_type()
 {
 	char file[] = "/sys/devices/virtual/dmi/id/product_name";
```

Comment:
`static const char`

Reply:
Sorry, I meant the variable `file`.  That way it is statically allocated at compile time rather than on the stack.

Reply:
Here's what I mean.  Using Compiler Explorer (https://godbolt.org/) the version with `static const` compiles to:
```
get_platform_type:
        pushq   %rbp
        movq    %rsp, %rbp
        subq    $16, %rsp
        movl    $.LC0, %esi
        movl    $file.0, %edi
        call    fopen
...

.LC0:
        .string "r"
file.0:
        .string "/sys/devices/virtual/dmi/id/product_name"
```

And the one without:
```
get_platform_type:
        pushq   %rbp
        movq    %rsp, %rbp
        subq    $64, %rsp
        movabsq $8531335224086852399, %rax
        movabsq $8244250539049968489, %rdx
        movq    %rax, -64(%rbp)
        movq    %rdx, -56(%rbp)
        movabsq $7596838301288461684, %rax
        movabsq $7237128952364689711, %rdx
        movq    %rax, -48(%rbp)
        movq    %rdx, -40(%rbp)
        movabsq $8459008537407743081, %rax
        movabsq $28549237878781027, %rdx
        movq    %rax, -39(%rbp)
        movq    %rdx, -31(%rbp)
        leaq    -64(%rbp), %rax
        movl    $.LC0, %esi
        movq    %rax, %rdi
        call    fopen
...

.LC0:
        .string "r"
```